spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerCheckinSpecTest.java

```diff
@@ -0,0 +1,209 @@
+        assertThat(consumer)
+            .isNotNull()
+            .doesNotReturn(initialCheckin, ConsumerDTO::getLastCheckin);
+    }
+
+    @Test
+    public void shouldNotUpdateLastCheckinWhenFetchingReleases() throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.random());
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+        OffsetDateTime initialCheckin = consumer.getLastCheckin();
+
+        consumerClient.consumers().getRelease(consumer.getUuid());
+
+        consumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+        assertThat(consumer)
+            .isNotNull()
+            .returns(initialCheckin, ConsumerDTO::getLastCheckin);
+    }
+
+    private HypervisorUpdateResultDTO hypervisorCheckin(OwnerDTO owner, ApiClient consumerClient,
+        String hypervisorName, String hypervisorId, List<String> guestIds, Map<String, String> facts,
+        String reporterId, boolean createMissing) throws ApiException, IOException {
+        JsonNode hostGuestMapping = getAsyncHostGuestMapping(hypervisorName, hypervisorId, guestIds, facts);
+        AsyncJobStatusDTO job = consumerClient.hypervisors().hypervisorUpdateAsync(
+            owner.getKey(), createMissing, reporterId, hostGuestMapping.toString());
+        HypervisorUpdateResultDTO resultData = null;
+        if (job != null) {
+            AsyncJobStatusDTO status = consumerClient.jobs().waitForJob(job.getId());
+            assertThatJob(status).isFinished();
+            resultData = getResultData(status);
+        }
+
+        return resultData;
+    }
+
+    private JsonNode getAsyncHostGuestMapping(String name, String id, List<String> expectedGuestIds,
+        Map<String, String> facts) {
+        ObjectMapper om = ApiClient.MAPPER;
+        ObjectNode root = om.createObjectNode();
+        ArrayNode hypervisors = root.putArray("hypervisors");
+        ObjectNode hypervisor = om.createObjectNode();
+        hypervisor.put("name", name);
+        ObjectNode hypervisorId = hypervisor.putObject("hypervisorId");
+        hypervisorId.put("hypervisorId", id);
+        ArrayNode guestIds = hypervisor.putArray("guestIds");
+        for (String gid : expectedGuestIds) {
+            ObjectNode guestId = om.createObjectNode();
+            guestId.put("guestId", gid);
+            guestIds.add(guestId);
+        }
+        JsonNode factsNode = om.valueToTree(facts);
+        hypervisor.set("facts", factsNode);
+        hypervisors.add(hypervisor);
+
+        return root;
+    }
+
+    private HypervisorUpdateResultDTO getResultData(AsyncJobStatusDTO status) throws IOException {
+        Map resultData = (Map) status.getResultData();
+        Gson gson = new GsonBuilder().create();
+        String json = gson.toJson(resultData);
```

Comment:
I'm not sure why I used Gson here, I might have seen this method in another spec test honestly. We should be using Jackson though. Updated

Reply:
Looks like I got this method from HypervisorCheckInSpecTest. It looks like there are two issues with deserializing the result data using Jackson. First is that the key and values are not in quotes, and this can be fixed by enabling the 
JsonReadFeature.ALLOW_UNQUOTED_FIELD_NAMES.mappedFeature(), but the second issue is that it uses equal signs instead of colons.

ResultData:
`{created=[{uuid=3b7c9add-66b6-4693-adab-5ccc708252ec, name=name-7BF284DA, owner={id=null, key=test_owner-E3171A71, displayName=null, href=null, contentAccessMode=null}}]`

Reply:
I apologize, I was passing in the `.toString()` value into the convertValue which didn't allow the de-serialization based on that String. 

` ApiClient.MAPPER.convertValue(status.getResultData().toString(), HypervisorUpdateResultDTO.class);`

Passing in the Object allows the `convertValue` to deserialize properly. Stupid mistake on my part *sigh* 

` ApiClient.MAPPER.convertValue(status.getResultData(), HypervisorUpdateResultDTO.class);`

I also removed the use of Gson in HypervisorCheckInSpecTest as well

Reply:
Interesting. Given that it's a string already, I wouldn't have expected that to make a difference.